bridge/build.rs

```diff
@@ -40,11 +33,106 @@ fn main() {
                 } else {
                     panic!("`solc` exited because it was terminated by a signal");
                 }
+            } else {
+                let output = Command::new("solc").args(&["--version"]).output().unwrap();
+                let output_string = String::from_utf8(output.stdout).unwrap();
+                let solc_version = output_string.lines().last().unwrap();
+                println!("cargo:rustc-env=SOLC_VERSION={}", solc_version);
             }
         }
         Err(err) => {
+            // we could have panicked here
             if let std::io::ErrorKind::NotFound = err.kind() {
-                panic!("`solc` executable not found in `$PATH`. `solc` is required to compile the bridge contracts. please install it: https://solidity.readthedocs.io/en/develop/installing-solidity.html");
+                // but let's see if solcjs is available or not
+                match Command::new("solcjs").arg("--version").output() {
+                    Ok(exit_status) => {
+                        let output_string = String::from_utf8(exit_status.stdout).unwrap();
+                        let solc_version = output_string.lines().last().unwrap();
+                        println!("cargo:rustc-env=SOLC_VERSION={}", solc_version);
+                        // overwrite option is not available in solcjs
+                        // so we better remove old compiled contracts
+                        match fs::remove_dir_all("../compiled_contracts") {
+                            Ok(()) => {
+                                println!("Removed old files");
+                            }
+                            Err(err) => {
+                                println!("Files not removed: {}", err);
```

Comment:
using `println!` for debugging is usually a bad pattern. Especially in `build.rs`